File: VB/CS/E911.Module.Web/Editors/WebCustomUserControlViewItem.vb
Imports Microsoft.VisualBasic
Imports System
Imports E911.Module.Editors
Imports System.ComponentModel
Imports DevExpress.ExpressApp.Web
Imports System.Collections.Generic
Imports DevExpress.ExpressApp.Model
Imports DevExpress.ExpressApp.Editors

Namespace E911.Module.Web.Editors
	Public Interface IModelWebCustomUserControlViewItem
	Inherits IModelCustomUserControlViewItem
		<Category("Data")> _
		Property UserControlPath() As String
	End Interface
	<ViewItem(GetType(IModelCustomUserControlViewItem))> _
	Public Class WebCustomUserControlViewItem
		Inherits CustomUserControlViewItem
		Protected model As IModelWebCustomUserControlViewItem
		Public Sub New(ByVal model As IModelViewItem, ByVal objectType As Type)
			MyBase.New(model, objectType)
			Me.model = TryCast(model, IModelWebCustomUserControlViewItem)
			If Me.model Is Nothing Then
				Throw New ArgumentNullException("IModelWebCustomUserControlViewItem must extend IModelCustomUserControlViewItem in the ExtendModelInterfaces method of your Web ModuleBase descendant.")
			End If
		End Sub
		Protected Overrides Function CreateControlCore() As Object
			'You can access the View and other properties here to additionally initialize your control.
			Return WebWindow.CurrentRequestPage.LoadControl(model.UserControlPath)
		End Function
	End Class
End Namespace

File: CS/CS/E911.Module.Web/Editors/WebCustomUserControlViewItem.cs
using System;
using E911.Module.Editors;
using System.ComponentModel;
using DevExpress.ExpressApp.Web;
using System.Collections.Generic;
using DevExpress.ExpressApp.Model;
using DevExpress.ExpressApp.Editors;

namespace E911.Module.Web.Editors {
    public interface IModelWebCustomUserControlViewItem : IModelCustomUserControlViewItem {
        [Category("Data")]
        string UserControlPath { get; set; }
    }
    [ViewItem(typeof(IModelCustomUserControlViewItem))]
    public class WebCustomUserControlViewItem : CustomUserControlViewItem {
        protected IModelWebCustomUserControlViewItem model;
        public WebCustomUserControlViewItem(IModelViewItem model, Type objectType)
            : base(model, objectType) {
            this.model = model as IModelWebCustomUserControlViewItem;
            if (this.model == null)
                throw new ArgumentNullException("IModelWebCustomUserControlViewItem must extend IModelCustomUserControlViewItem in the ExtendModelInterfaces method of your Web ModuleBase descendant.");
        }
        protected override object CreateControlCore() {
            //You can access the View and other properties here to additionally initialize your control.
            return WebWindow.CurrentRequestPage.LoadControl(model.UserControlPath);
        }
    }
}